tests/unit_tests/test_cell.py

```diff
@@ -112,14 +123,118 @@ def test_get_nuclides(uo2):
     assert nucs == ['U235', 'O16']
 
 
+def test_volume_setting():
+    c = openmc.Cell()
+
+    # Test ordinary volume and uncertain volume
+    c.volume = 3
+    c.volume = u.ufloat(3, 0.7)
+
+    # Allow volume to be set to 0.0
+    c.volume = 0.0
+    c.volume = u.ufloat(0.0, 0.1)
+
+    # Test errors for -ve volume
+    with pytest.raises(ValueError):
+        c.volume = -1.0
+    with pytest.raises(ValueError):
+        c.volume = u.ufloat(-0.05, 0.1)
+
+
+def test_atoms_material_cell(uo2, water):
+    """ Test if correct number of atoms is returned.
+    Also check if Cell.atoms still works after volume/material was changed
+    """
+    c = openmc.Cell(fill=uo2)
+    c.volume = 2.0
+    expected_nucs = ['U235', 'O16']
+
+    # Precalculate the expected number of atoms
+    M = (atomic_mass('U235') + 2 * atomic_mass('O16')) / 3
+    expected_atoms = list()
+    expected_atoms.append(1/3 * uo2.density/M * AVOGADRO * 2.0)  # U235
+    expected_atoms.append(2/3 * uo2.density/M * AVOGADRO * 2.0)  # O16
+
+    tuples = list(c.atoms.items())
```

Comment:
If you're just iterating over `items()`, no need to convert it to a list explicitly
```suggestion
    tuples = c.atoms.items()
```